src/components/application_manager/src/request_controller.cc

```diff
@@ -364,9 +369,27 @@ void RequestController::onTimer() {
   SDL_AUTO_TRACE();
   SDL_DEBUG(
       "ENTER Waiting fore response count: " << waiting_for_response_.Size());
-  RequestInfoPtr probably_expired =
-      waiting_for_response_.FrontWithNotNullTimeout();
-  while (probably_expired && probably_expired->isExpired()) {
+  while(!timer_stop_flag_) {
```

Comment:
@LuxoftAKutsan The while condition can also be stopped by ```break``` besides in destructor, 
This purpose is to keep sure whenever this onTimer is called,  the current thread both under condition ```!expired``` and under condition ```!probably_expired->isExpired()``` can be blocked to wake up(previous lack of this purpose). Other executed logic is same with original condition ```(probably_expired && probably_expired->isExpired())```. I am not very clearly understand your thread logic